client/my-sites/themes/index.web.js

```diff
@@ -4,25 +4,17 @@
 import config from 'config';
 import userFactory from 'lib/user';
 import { makeLayout } from 'controller';
-import { navigation, siteSelection } from 'my-sites/controller';
+import { makeNavigation, siteSelection } from 'my-sites/controller';
 import { singleSite, multiSite, loggedOut } from './controller';
 
-// FIXME: These routes will SSR the logged-out Layout even if logged-in.
-// While subsequently replaced by the logged-in Layout on the client-side,
-// we'll want to render it on the server, too.
-
-// `logged-out` middleware isn't SSR-compliant yet, but we can at least render
-// the layout.
-// FIXME: Also create loggedOut/multiSite/singleSite elements, depending on route.
-
 export default function( router ) {
 	const user = userFactory();
 	const isLoggedIn = !! user.get();
 
 	if ( config.isEnabled( 'manage/themes' ) ) {
 		if ( isLoggedIn ) {
-			router( '/design/:tier(free|premium)?', multiSite, navigation, siteSelection );
-			router( '/design/:tier(free|premium)?/:site_id', singleSite, navigation, siteSelection );
+			router( '/design/:tier(free|premium)?', siteSelection, multiSite, makeNavigation, makeLayout );
```

Comment:
I'm not entirely sure TBH :-D `siteSelection` is... a mouthful, and it selects the current site with some fallback to all sites if none are selected, and then some. Anyway, I've removed the mw from the multisite route (1077401), and it looks like things are still working.